generated/nidaqmx/_grpc_interpreter.py

```diff
@@ -3356,15 +3357,15 @@ def write_digital_u8(
             grpc_types.WriteDigitalU8Request(
                 task=task, num_samps_per_chan=num_samps_per_chan,
                 auto_start=auto_start, timeout=timeout,
-                data_layout_raw=data_layout, write_array=write_array))
+                data_layout_raw=data_layout, write_array=bytes(write_array)))
```

Comment:
This does not validate that write_array has the correct dtype. If the array has the wrong dtype, this writes the wrong data.

LibraryInterpreter validates the array's dtype by specifying it on the ndpointer_wrapper.
Example code:
```
import nidaqmx
import numpy
from nidaqmx.constants import LineGrouping
from nidaqmx.stream_writers import DigitalSingleChannelWriter

with nidaqmx.Task() as task:
    task.do_channels.add_do_chan(
        "Dev1/port1/line0:3", line_grouping=LineGrouping.CHAN_FOR_ALL_LINES
    )

    writer = DigitalSingleChannelWriter(task.out_stream, auto_start=True)
    write_array = numpy.array([1, 2, 4, 8], dtype=numpy.uint8)
    print(writer.write_many_sample_port_byte(write_array))
```

With the correct dtype of uint8, it writes 4 bytes.

With the incorrect dtype of uint32, it raises an exception:
```
PS D:\dev\nidaqmx-python> poetry run python .\examples\do_sw_timed.py
Traceback (most recent call last):
  File "D:\dev\nidaqmx-python\examples\do_sw_timed.py", line 14, in <module>
    print(writer.write_many_sample_port_byte(write_array))
  File "D:\dev\nidaqmx-python\generated\nidaqmx\stream_writers.py", line 836, in write_many_sample_port_byte
    return self._interpreter.write_digital_u8(
  File "D:\dev\nidaqmx-python\generated\nidaqmx\_library_interpreter.py", line 5850, in write_digital_u8
    error_code = cfunc(
ctypes.ArgumentError: argument 6: <class 'TypeError'>: array must have data type uint8
```

GrpcStubInterpreter is using `bytes(write_array)`. With a dtype of uint32, `print(bytes(write_array))` prints a lot of zero bytes:
```
b'\x01\x00\x00\x00\x02\x00\x00\x00\x04\x00\x00\x00\x08\x00\x00\x00'
```

NI IO Trace shows that this writes the zero bytes:
```
177681.  DAQmxWriteDigitalU8 (0x00007FFBD523CD28, 4 (0x4), True, 10.000000 (1.000000E+01), "DAQmx_Val_GroupByChannel", {0x01,0x00,...}, 4 (0x4), NULL)
Process ID: 0x0000246C         Thread ID: 0x00003F98
Start Time: 14:29:39.2682      Call Duration 00:00:00.0000
Status: 0 (0x0)
```
![image](https://github.com/ni/nidaqmx-python/assets/17914399/f0a453ba-962d-4bda-8bb2-54d2c35842f9)

Reply:
This affects a lot more than these three functions, so I filed an internal bug: AB#2390189: GrpcStubInterpreter reads/writes invalid data when NumPy array has wrong dtype